engine/sources/local/source.go

```diff
@@ -184,7 +186,18 @@ func (ls *localSourceHandler) Snapshot(ctx context.Context, g session.Group) (bk
 	}
 
 	timeoutCtx, cancel := context.WithCancelCause(ctx)
-	timeoutCtx, _ = context.WithTimeoutCause(timeoutCtx, 5*time.Second, fmt.Errorf("timeout: %w", context.DeadlineExceeded))
+	timeout := 5
+	envTimeoutStr := os.Getenv("SNAPSHOT_CONTEXT_TIMEOUT")
```

Comment:
I think if we want to configure the snapshot context timeout  at some point, this should be part of the [engine configuration](https://docs.dagger.io/reference/configuration/engine/) through `engine.[json|toml]` instead of being fetch from an environment variable.

However, I wouldn't recommend working on that changes until we know if that something we want to do or not.